### ISSUE TITLE
Tests from external dependencies must be excluded from test catalog.

### ISSUE THREAD

**Issue by arrowplum** [NumberFour/n4js#351]
We have started getting tests from `Mangelhaft` in the test catalog for OPR. 

This is a regression, although I cannot find the old ticket that covered this.

NOTE: The tests in the test catalog must really not include any tests that are in the external dependencies. It is not enough to remove the tests from those packages.

**Comment by yoosiba** [comment]
Seems like commit ce12cd2 updated libraries in the wrong way.

@arrowplum @dbo if we filter out tests from the externals there is no way one can distribute test for the implementation as separate artifacts. The only way to do this will be to clone given test project into the workspace of the tested project. Whole purpose of the libraries manager is for the users not to be burdened with cloning/importing their dependencies into the workspace.

Do you want really want IDE and N4JSC.jar to filter out test files from provided projects?
Or should I adjust provided projects (essentially fix broken merge that was mentioned) and keep behaviour we had so far?

**Comment by arrowplum** [comment]
Let me see if I am misunderstanding @yoosiba. My understanding was that the library manager lets you add dependencies that you use in your project. Your project should be tested against whatever dependencies you define, but the tests of the dependencies themselves should not be run in the context of your project.

In other words, if for some reason react used mangelhaft, we should not run react's tests because we are using that library. Or if online presence used stdlib, we shouldn't run stdlib's tests when we are testing onlinepresence. stdlib has its own test, own build cycle, own everything. 

In other words we don't test the things we use, we only test how we use them. That is my understanding.

**Comment by yoosiba** [comment]
I see. what you are saying @arrowplum  But if I have API and API-TEST projects in separate npms. In my workspace I have API-IMPL that I am working on. Wouldn't I want to run those API-TESTs to verify if my IMPL is correct?

**Comment by arrowplum** [comment]
Normally you wouldn't have your impl have a dependency on your tests, it would be the other way around.

In our nonstandard api-impl split, you still wouldnt want to run the tests from an external npm. You would load the projects together in the ide.

**Comment by yoosiba** [comment]
I guess you are right. eot

**Comment by yoosiba** [comment]
To whoever might be concerned, on master we have removed test bundles from provided libraties. Also I am adding script that will be updating those on IDE side - should prevent accidental additions. 

Active filtering of the tests from LibraryManager is still TBD.

**Comment by mmews-n4** [comment]
continued here https://github.com/eclipse/n4js/issues/555
